tools/wptrunner/wptrunner/testloader.py

```diff
@@ -619,10 +619,30 @@ def new_group(self,
         state["prev_group_key"] = (subsuite, test_type, path)
         return rv
 
+    def in_one_group(self,
+                     subsuite: str,
+                     tests: List[wpttest.Test]) -> bool:
+        small_subsuite_size = self.kwargs.get("small_subsuite_size", 0)
+        return len(subsuite) > 0 and len(tests) <= small_subsuite_size
+
     def make_groups(self, tests_by_type: TestsByType) -> List[TestGroup]:
         groups = []
         state: MutableMapping[str, Any] = {}
         for (subsuite, test_type), tests in tests_by_type.items():
+            # For subsuites only have few tests, put them in one group so that
+            # it will be picked up by one worker and only one restart will be
+            # needed. Tests have a different test type will still be put into
+            # different group though.
+            if self.in_one_group(subsuite, tests):
+                state["prev_group_key"] = (subsuite, test_type, [""])
+                group_metadata = self.group_metadata(state)
+                groups.append(TestGroup(deque(), subsuite, test_type, group_metadata))
+                for test in tests:
+                    group, _, _, metadata = groups[-1]
+                    group.append(test)
+                    test.update_metadata(metadata)
```

Comment:
This bit is exactly like the loop below. Can you make the entire bit something like:

```py
in_one_group = self.in_one_group(subsuite, tests)
if in_one_group:
    state["prev_group_key"] = (subsuite, test_type, [""])
    group_metadata = self.group_metadata(state)
    groups.append(TestGroup(deque(), subsuite, test_type, group_metadata))

for test in tests:
    if not in_one_group and self.new_group(state, subsuite, test_type, group_metadata):
        group_metadata = self.group_metadata(state)
        groups.append(TestGroup(deque(), subsuite, test_type, group_metadata))

    group, _, _, metadata = groups[-1]
    group.append(test)
    test.update_metadata(metadata)
````